### DISCUSSION TITLE
onMessage does not trigger after restarting the app

### DISCUSSION THREAD

**Discussion by zackyrza** [invertase/react-native-firebase#5252] [Q&A] [answered]
I don't know what happen with my project, but the cloud messaging and notification is perfectly work just on the first build and first running the app. After I kill the app and run it again, the onMessage handler not triggered when I send test notification from firebase console. But the notification does appear and when I click it, it just open the app and nothing happened. I have tried to create new clean project from scratch and it is working like a charm.

The problem is only occurred in my project only, does anyone have a same problem like me?

**Comment by zackyrza** [comment, marked as answer]
never mind, solved by downgrading `@react-native-firebase/messaging` to version `7.4.1`

**Reply by mikehardy** [reply]
That is not a solution and is certainly no way forward, you can't even use firebase-ios-sdk v7.x.x much less the new v8.x.x coming out soon. v7.x has things needed for ios14 IIRC. It's open source so do what you like, but I'd try to figure out your actual root cause with modern versions
